scripts/setup-wsl.ps1

```diff
@@ -0,0 +1,156 @@
+        [string] $Separator = ';',
+
+        [Parameter(ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [switch] $Global
+    )
+
+    if (-not $IsWinEnv) {
+        return
+    }
+
+    if (-not $Global) {
+        [string] $Environment = 'Environment'
+        [Microsoft.Win32.RegistryKey] $Key = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey($Environment, [Microsoft.Win32.RegistryKeyPermissionCheck]::ReadWriteSubTree)
+    } else {
+        [string] $Environment = 'SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
+        [Microsoft.Win32.RegistryKey] $Key = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey($Environment, [Microsoft.Win32.RegistryKeyPermissionCheck]::ReadWriteSubTree)
+    }
+
+    # $key is null here if it the user was unable to get ReadWriteSubTree access.
+    if ($null -eq $Key) {
+        throw (new-object -typeName 'System.Security.SecurityException' -ArgumentList "Unable to access the target registry")
+    }
+
+    # Keep current ValueKind if possible/appropriate
+    try {
+        [Microsoft.Win32.RegistryValueKind] $ValueKind = $Key.GetValueKind($Name)
+    } catch {
+        [Microsoft.Win32.RegistryValueKind] $ValueKind = [Microsoft.Win32.RegistryValueKind]::String
+    }
+
+    if ($Append) {
+        # Get current unexpanded value
+        [string] $CurrentUnexpandedValue = $Key.GetValue($Name, '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
+
+        # Evaluate new value
+        $NewValue = [string]::Concat($CurrentUnexpandedValue.TrimEnd($Separator), $Separator, $Value)
+    } else {
+        $NewValue = $Value
+    }
+
+    # Upgrade ValueKind to [Microsoft.Win32.RegistryValueKind]::ExpandString if appropriate
+    if ($NewValue.Contains('%')) {
+        $ValueKind = [Microsoft.Win32.RegistryValueKind]::ExpandString
+    }
+
+    $Key.SetValue($Name, $NewValue, $ValueKind)
+}
+
+
+
+# Setting Tls to 12 to prevent the Invoke-WebRequest : The request was
+# aborted: Could not create SSL/TLS secure channel. error.
+$originalValue = [Net.ServicePointManager]::SecurityProtocol
+[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor 3072 # = [Net.SecurityProtocolType]::Tls12
+
+try {
+    # Check if mkcert is installed
+    try {
+        mkcert -version
+    } catch [System.Management.Automation.CommandNotFoundException] {
+        # Check if Chocolatey is installed
+        try {
+            choco
+        } catch {
+            # Chocolatey was not found, install it now
+            #Invoke-Expression ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+            Invoke-Expression "& { $(Invoke-RestMethod 'https://chocolatey.org/install.ps1') }"
+        }
+
+        # Install mkcert
+        choco install mkcert -y
+    }
+
+    # Setup mkcert
+    mkcert -install
+
+    # Set default WSL version
+    wsl --set-default-version 2
```

Comment:
Throws an exception if WSL isn't installed. So we have to catch it and show a meaningful warning to the user.